cordapp/src/main/kotlin/net/corda/plugins/CordappPlugin.kt

```diff
@@ -156,8 +156,8 @@ class CordappPlugin : Plugin<Project> {
         return Pair(targetPlatformVersion, minimumPlatformVersion)
     }
 
-    private fun parseVersion(versionStr: String?): Int {
-        if (versionStr == null)
+    private fun parseVersion(versionStr: Any?): Int {
+        if (versionStr == null || versionStr !is String)
```

Comment:
_**AAAGH!** I'm so confused!_ :angry: 

This `versionId` is _documented_ as being `Any` type because
> relaxed type so users can specify Integer or String identifiers

which implies that we would also like to allow something like `versionId = '1-Potato'`.

So I'd have thought that we'd want something more like:
```kotlin
attributes["Cordapp-Contract-Version"] = cordapp.contract.versionId.let { v -> v as? CharSequence ?: parseVersion(v) }
```
and then
```kotlin
private fun parseVersion(versionStr: Any?): Int {
    if (versionStr == null)
        throw InvalidUserDataException("Target versionId not specified. Please specify a whole number starting from 1.")
    return try {
        val version = Integer.parseInt(versionStr.toString())
        ...
```
Although these error messages aren't consistent with that philosophy either :confused:.

And *YES*, we do need `as? CharSequence` instead of `as? String` because someone might pass Gradle a Groovy `GString` here by using double-quotes :man_facepalming:. E.g.
```gradle
cordapp {
    contract {
        versionId "$cordappContractVersion_info_arg"
```

Some extra test cases would be nice...

Reply:
If `versionId` were a `String` then Groovy would convert the `GString` to `String` automatically. But we're using `Any` so that someone can also use an `Int`.

Reply:
yes - it was meant to cater for Int or String specifications (and nothing else).
And these changes are not even related to this PR so I'm going to rollback the last commit and leave it there.

Reply:
OK, that new JIRA will be with you momentarily... :imp: